sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala

```diff
@@ -280,6 +280,20 @@ class SQLQuerySuite extends QueryTest with SharedSparkSession with AdaptiveSpark
     )
   }
 
+  test("SPARK-43522: Fix creating struct column name with index of array") {
+
+    val data = Seq("a=b,c=d,d=f").toDF.withColumn("key_value", split('value, ","))
+
+    val df = data.withColumn("map_entry", transform(col("key_value"), x => struct(split(x, "=")
```

Comment:
Could you elaborate on the difference between the two DataFrames? I think it would be easier to compare to a DataFrame with the expected result.

Reply:
The difference are one use alias to struct elements, another one don't use alias. The one use alias will get rigt answer before this PR, another will not. I move it from jira ticket.

> I think it would be easier to compare to a DataFrame with the expected result.

Just check expected result is good for me. I will change it